lib/java/desktop-environment/src/main/java/org/enso/desktopenvironment/LinuxTrashBin.java

```diff
@@ -22,13 +22,24 @@
  * for each trashed entry in the files directory.
  */
 final class LinuxTrashBin implements TrashBin {
-
+  private static LinuxTrashBin instance;
+  private final LinuxDirectories directories;
   private static final String XDG_DATA_HOME = "XDG_DATA_HOME";
   private static final String PATH_TRASH = "Trash";
   private static final String PATH_FILES = "files";
   private static final String PATH_INFO = "info";
 
-  private final LinuxDirectories directories = new LinuxDirectories();
+  public static LinuxTrashBin getInstance() {
```

Comment:
At least _encapsulate_:
```suggestion
  static LinuxTrashBin getInstance() {
```
but probably do way more. I really don't think this _lazy initialization pattern_ is good for anything!

Reply:
Done in a99333153fded4a2ef03498d85bdb9c05fb6fca4 and 6951fd305322e86d009af82181ca101051fb3cf3